File: files/CSEC 19.05.py
import pandas as pd

input = pd.read_excel("files/Excel Challenge 19th May.xlsx", usecols="B:E", skiprows=1, nrows=10)
test = pd.read_excel("files/Excel Challenge 19th May.xlsx", usecols= "G", skiprows=1, nrows=4)

result = input.assign(Combined=input.iloc[:, 1:4].apply(lambda x: ', '.join(sorted(x.dropna())), axis=1))
result = result.assign(n=result.groupby('Combined')['Combined'].transform('size')).query('n == 1').loc[:, ['Product']].reset_index(drop=True)
result.columns = ['Products']

print(result.equals(test))

File: files/2024-05-19/CSEC 19.05.R
library(tidyverse)
library(readxl)

input = read_excel("files/Excel Challenge 19th May.xlsx", range = "B2:E11")
test  = read_excel("files/Excel Challenge 19th May.xlsx", range = "G2:G6")

result = input %>%
  mutate(Combined = pmap_chr(.[2:4], ~ toString(sort(c(...))))) %>%
  mutate(n = n(), .by = Combined) %>%
  filter(n == 1) %>%
  select(Products = Product)

identical(result, test)
#> [1] TRUE